pkg/iptables/iptables.go

```diff
@@ -105,7 +127,14 @@ func NewIPv6Rule(table, chain string, spec ...string) Rule {
 	return &rule{table, chain, spec, ProtocolIPv6}
 }
 
-func (r *rule) Add(client Client) error {
+func (r *rule) Prepend(client Client) error {
+	if err := client.Insert(r.table, r.chain, 1, r.spec...); err != nil {
```

Comment:
@clive-jevons should we maybe use a `PrependUnique` here?

Reply:
I _think_ the intention of the `Client` interface was to only shadow the existing API of the ipsets client? As such the `InsertUnique` semantics should probably be part of the API here? Could be wrong though - some guidance by @squat or @leonnicolas would be good here, I think 😬

Reply:
If you look a couple of lines down the `Append()` function is implemented via `AppendUnique()` - hence the question.

I think this is relevant when kilo was terminated abruptly without having a chance to clean up and then spins up again - without the check for `Exists()` the `Insert()` would probably cause an error?

Reply:
Very fair point 😉 I'd still be interested in the feedback / guidance from the maintainers on this 😬 
But absolutely agree with the necessity of the functionality - where it ends up being implemented 👍

Reply:
You're right re the interface - it's implemented in an external dependency which doesn't mirror `AppendUnique()` for `Exists()`. So my suggestion here would be to implement this here while at the same time making a PR in the dependency to implement `ExistsUnique()` and later switch to it once it's there. :thinking:

Reply:
done

Reply:
Some good news here: the [go-iptables PR](https://github.com/coreos/go-iptables/pull/92) has moved an initial step forward.

Reply:
❤️

Reply:
Progress here: it's been merged to main. Once there is a new release we can make use of the new function.

Reply:
Great work!

Reply:
Hmm we can probably just pin to the latest commit on main rather than wait for a release, no?

Reply:
have done that and removed all references to `iptables.Insert()`